tests/infrastructure/src/main/java/org/mule/test/infrastructure/process/rules/MuleDeployment.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.infrastructure.process.rules;
+
+import static java.lang.Boolean.parseBoolean;
+import static java.lang.Integer.parseInt;
+import static java.lang.System.getProperty;
+import static org.apache.commons.io.FileUtils.deleteDirectory;
+import static org.apache.commons.io.FileUtils.deleteQuietly;
+import static org.apache.commons.io.FileUtils.moveDirectory;
+import static org.apache.commons.io.FilenameUtils.removeExtension;
+import static org.apache.commons.lang.StringUtils.isEmpty;
+import static org.mule.test.infrastructure.process.AppDeploymentProbe.isDeployed;
+import org.mule.tck.probe.PollingProber;
+import org.mule.test.infrastructure.process.MuleProcessController;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang.StringUtils;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * JUnit rule to deploy a Mule application for testing in a local Mule server. Usage:
+ * <p>
+ * <pre>
+ * public static class MuleDeployment {
+ *  &#064;Rule
+ *  public static MuleDeployment deployment = application(&quot;/path/to/application.zip&quot;).withProperty("-M-Dproperty", "value").timeout(120).deploy();
+ *
+ *  &#064;Test
+ *  public void useApplication() throws IOException {
+ *      // App is deployed
+ *      // This code exercises the application
+ *  }
+ * }
+ * </pre>
+ */
+public class MuleDeployment implements TestRule
+{
+
+    private static final String DISTRIBUTION_PROPERTY = "com.mulesoft.muleesb.distributions:mule-ee-distribution-standalone:zip";
+    private static final long POLL_DELAY_MILLIS = 1000;
+    private static Logger logger = LoggerFactory.getLogger(MuleDeployment.class);
+    private static final File WORKING_DIRECTORY = new File(getProperty("user.dir"));
+    private static final String DELETE_ON_EXIT = getProperty("mule.test.deleteOnExit");
+    private static final int DEFAULT_DEPLOYMENT_TIMEOUT = 60000;
+    private int deploymentTimeout = getProperty("mule.test.deployment.timeout") == null ? DEFAULT_DEPLOYMENT_TIMEOUT : parseInt(getProperty("mule.test.deployment.timeout"));
```

Comment:
You can use the getProperty() with the second attribute as the default value and avoid calling the getProperty() twice. In this case it will be parseInt( getProperty("mule.test.deployment.timeout", DEFAULT_DEPLOYMENT_TIMEOUT) )